File: amazon-sqs-email-poc/src/main/java/com/moneyflow/flow/service/EmailServiceVerification.java
package com.moneyflow.flow.service;

import com.amazonaws.services.simpleemail.AmazonSimpleEmailService;
import com.amazonaws.services.simpleemail.model.VerifyEmailAddressRequest;
import com.moneyflow.flow.dto.EmailStructureDTO;
import com.moneyflow.flow.enums.TypeMail;
import com.moneyflow.flow.service.imp.EmailSendResolver;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Service
@Slf4j
@RequiredArgsConstructor
public class EmailServiceVerification implements EmailSendResolver {

    private final AmazonSimpleEmailService amazonSimpleEmailService;

    @Override
    public TypeMail getTypeMail() {
        return TypeMail.VERIFICATION;
    }

    public void send(final EmailStructureDTO emailStructure) {

        log.info("Iniciando envio de e-mail pelo AWS SES com a estrutura -> {}", emailStructure);

        final VerifyEmailAddressRequest aws = emailStructure.toAwsVerify();

        amazonSimpleEmailService.verifyEmailAddress(aws);
    }

}

File: server/aws-lambda-ses/service/email_service_verification.go
package service

import (
	"github.com/aws/aws-sdk-go/service/ses"
	"golang/service/imp"
	_struct "golang/struct"
)

const TypeVerification = "VERIFICATION"

type EmailServiceVerification struct {
	emailService *ses.SendEmailInput
}

func (e *EmailServiceVerification) Type(typeMail string) bool {
	return typeMail == TypeVerification
}

func (e *EmailServiceVerification) Send(dto _struct.EmailStructureDTO) {

	verifyAddress := dto.ToAwsVerify()

	session := imp.Initialize()

	_, err := session.VerifyEmailIdentity(verifyAddress)
	if err != nil {
		panic("Error to send verification mail")
	}
}
